inc/formatconvert.class.php

```diff
@@ -2310,21 +2143,22 @@ static function printerInventoryTransformation($array) {
     * @return string "Drive" or "HardDrive"
     */
    static function getTypeDrive($data) {
-      if (((isset($data['TYPE'])) AND
-              ((preg_match("/rom/i", $data["TYPE"])) OR (preg_match("/dvd/i", $data["TYPE"]))
-               OR (preg_match("/blue.{0, 1}ray/i", $data["TYPE"]))))
-            OR
-         ((isset($data['MODEL'])) AND
-              ((preg_match("/rom/i", $data["MODEL"])) OR (preg_match("/dvd/i", $data["MODEL"]))
-               OR (preg_match("/blue.{0, 1}ray/i", $data["MODEL"]))))
-            OR
-         ((isset($data['NAME'])) AND
-              ((preg_match("/rom/i", $data["NAME"])) OR (preg_match("/dvd/i", $data["NAME"]))
-               OR (preg_match("/blue.{0, 1}ray/i", $data["NAME"]))))) {
-
-         return "Drive";
-      } else {
-         return "HardDrive";
+      $to_match_regex = ['rom', 'dvd', 'blue.{0,1}ray', 'reader',
```

Comment:
I do not know what the agent will give as data; but the name is blu-ray, not blue-ray.

I guess you should use something like `(\s-)?` (not 100% sure, would need to be tested, but you got the idea) rather than `{0, 1}` to handle multiple chars cases.